test_app/settings.py

```diff
@@ -15,225 +19,33 @@
-    'cow_moo': {'shortname': 'action_base', 'model_name': 'test_app.cow', 'name': 'Cow Mooer', 'action': 'say_cow'},
-}
-ANSIBLE_BASE_JWT_MANAGED_ROLES.append("System Auditor")  # noqa: F821 this is set by dynamic settings for jwt_consumer
-ANSIBLE_BASE_ALLOW_SINGLETON_USER_ROLES = True
-ANSIBLE_BASE_ALLOW_SINGLETON_TEAM_ROLES = True
-ANSIBLE_BASE_RBAC_MODEL_REGISTRY = {
-    "test_app.inventory": {"parent_field_name": "organization"},
-    "test_app.credential": {},
-    "test_app.immutabletask": {"parent_field_name": None},
-}
-ANSIBLE_BASE_OAUTH2_PROVIDER_PERMISSIONS_CHECK_IGNORED_VIEWS = ["drf_spectacular.views.SpectacularSwaggerView"]
-ALLOW_SHARED_RESOURCE_CUSTOM_ROLES = True  # Allow making custom roles with org change permission, for example
-ALLOW_LOCAL_ASSIGNING_JWT_ROLES = False
-
-ANSIBLE_BASE_USER_VIEWSET = 'test_app.views.UserViewSet'
-
-LOGIN_URL = "/login/login"
-
-RESOURCE_SERVER = {
-    "URL": "http://localhost",
-    "SECRET_KEY": "my secret key",
-    "VALIDATE_HTTPS": False,
-}
-RESOURCE_SERVICE_PATH = "/api/v1/service-index/"
-# Backwards sync turned off, because for most of the duration of the tests
-# the resource server will not actually be running
-# so it will be flipped true only for specific tests that test this
-RESOURCE_SERVER_SYNC_ENABLED = False
-
-RENAMED_USERNAME_PREFIX = "dab:"
-
-FLAGS = {
-    "FEATURE_SOME_PLATFORM_FLAG_ENABLED": [
-        {
-            "condition": "boolean",
-            "value": False,
-            "required": True,
-        },
-        {
-            "condition": "before date",
-            "value": '2022-06-01T12:00Z',
-        },
-    ],
-    "FEATURE_SOME_PLATFORM_FLAG_FOO_ENABLED": [
-        {
-            "condition": "boolean",
-            "value": False,
-        },
-    ],
-    "FEATURE_SOME_PLATFORM_FLAG_BAR_ENABLED": [
-        {
-            "condition": "boolean",
-            "value": True,
-        },
-    ],
-}
+# Create a the standard DYNACONF instance which will come with DAB defaults
+# this loads the files passed to `settings_files` list
+# and environment specific file e.g: development_{filename}.py
+# assuming the current env is development and files are located in the same folder
+DYNACONF = factory(
+    __name__,
+    "TESTAPP",
+    environments=("development", "sqlite"),
+    settings_files=["defaults.py"],
+)
+
+# Load new standard settings files from
+#  /etc/ansible-automation-platform/ and /etc/ansible-automation-platform/testapp/
+load_standard_settings_files(DYNACONF)
+
+# Set overrides that must be set after DAB and file settings are loaded
+DYNACONF.set(
+    "ANSIBLE_BASE_JWT_MANAGED_ROLES",
+    "@merge System Auditor",
+    loader_identifier="add_system_auditor",
+)
+
+# Load envvars at the end to allow them to override everything loaded so far
+load_envvars(DYNACONF)
+
+# Update django.conf.settings with DYNACONF keys.
+export(__name__, DYNACONF)
+
+# Validate the settings according to the validators registered
+validate(DYNACONF)
```

Comment:
Other than the `.set` is there anyway we can "just do this" when creating the factory?

Reply:
You mean, defer the validation when creating the object?

we can add the validation as a post load hook, that is a great idea, so we enforce validation.

NOTE: Right now there is no validator declared yet.